cmd/cluster-init/update_bootstrap_secrets.go

```diff
@@ -0,0 +1,348 @@
+package main
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/api/secretbootstrap"
+)
+
+const (
+	credentials         = "credentials"
+	regPullCredsAll     = "registry-pull-credentials-all"
+	dotDockerConfigJson = ".dockerconfigjson"
+	arm01               = "arm01"
+	testCredentials     = "test-credentials"
+	kubeconfig          = "kubeconfig"
+	config              = "config"
+)
+
+func updateCiSecretBootstrap(o options) error {
+	secretBootstrapDir := filepath.Join(o.releaseRepo, "core-services", "ci-secret-bootstrap")
+	secretBootstrapConfigFile := filepath.Join(secretBootstrapDir, "_config.yaml")
+	logrus.Infof("Updating ci-secret-bootstrap: %s", secretBootstrapConfigFile)
+
+	var c secretbootstrap.Config
+	if err := secretbootstrap.LoadConfigFromFile(secretBootstrapConfigFile, &c); err != nil {
+		return err
+	}
+	if err := updateCiSecretBootstrapConfig(o, &c); err != nil {
+		return err
+	}
+	return secretbootstrap.SaveConfigToFile(secretBootstrapConfigFile, &c)
+}
+
+func updateCiSecretBootstrapConfig(o options, c *secretbootstrap.Config) error {
+	for _, groupName := range []string{buildUFarm, "non_app_ci", "non_app_ci_x86"} {
+		c.ClusterGroups[groupName] = append(c.ClusterGroups[groupName], o.clusterName)
+	}
+	c.UserSecretsTargetClusters = append(c.UserSecretsTargetClusters, o.clusterName)
+	if err := updatePodScalerSecret(c, o); err != nil {
```

Comment:
Is there a common interface for these like you have for the top-level steps that `cluster-init` takes?

Reply:
I was thinking that might be useful, since you also mentioned it I will refactor it.